test/test_kubeclient.rb

```diff
@@ -440,10 +460,13 @@ def test_api_basic_auth_back_comp_success
       auth_options: { user: 'username', password: 'password' }
     )
 
-    pods = client.get_pods
+    pods = []
+    client.get_pods do |pod|
+      pods << pod
+    end
 
-    assert_equal('Pod', pods.kind)
     assert_equal(1, pods.size)
+    assert_equal('Kubeclient::Pod', pods[0].class.to_s)
```

Comment:
a kubernetes `PodList` object has a field `Kind` which is essentially its class name. Since we're no longer returning a `PodList` (if `block_given?`), `Kind` is no longer available to us. Since i removed the line `assert_equal('Pod', pods.kind)`,  I wanted to preserve the logic.